pkg/aggregator/aggregator.go

```diff
@@ -185,78 +185,63 @@ func mergeSpecs(dest, source *spec.Swagger, renameModelConflicts, ignorePathConf
 			source = FilterSpecByPathsWithoutSideEffects(source, keepPaths)
 		}
 	}
-	// Check for model conflicts
-	conflicts := false
+
+	// Check for model conflicts and rename to make definitions conflict-free (modulo different GVKs)
+	usedNames := map[string]bool{}
+	for k := range dest.Definitions {
+		usedNames[k] = true
+	}
+	renames := map[string]string{}
+DEFINITIONLOOP:
 	for k, v := range source.Definitions {
-		v2, found := dest.Definitions[k]
-		if found && !deepEqualDefinitionsModuloGVKs(&v2, &v) {
-			if !renameModelConflicts {
-				return fmt.Errorf("model name conflict in merging OpenAPI spec: %s", k)
-			}
-			conflicts = true
-			break
+		existing, found := dest.Definitions[k]
+		if !found || deepEqualDefinitionsModuloGVKs(&existing, &v) {
+			// skip for now, we copy them after the rename loop
+			continue
 		}
-	}
 
-	if conflicts {
```

Comment:
I removed this optimization (if it ever has been one, am not so sure) because all of our specs have conflicts (due to types like Status, DeleteOptions, ...).